### PR TITLE
Fix (some) multiple chooser search issues

### LINKED ISSUE #10451: MultipleChooserPanel lose checkbox on search
`MultipleChooserPanel("companies", chooser_field_name="company", label=_("Companies")),
`
Above is a pointing to a page. The checkboxes are to make the checkbox select dissapear if you use the search field. This renders the multipagechooser useless at the moment. This only happens when you relate to page, snippets and other models are fine.

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project:  no, not on a fresh install

### Technical details

-   Python version: Run`python 3.10
-   Django version: 4.2
-   Wagtail version: 5.0
-   Browser version: chrome latest

### PR DESCRIPTION
**Pull request by tomkins**
Fixes #10451 (I know this one was closed as a duplicate of #10754, but it feels like there's multiple issues)
Partial fixes towards #10754

### To test:
Feel free to use https://github.com/tomkins/multichooserdemo/tree/page-chooser-demo as a demo project. Quick script to create a bunch of Standard Pages in that demo:

```
from home.models import HomePage, StandardPage
home_page = HomePage.objects.get()

for i in range(100):
    home_page.add_child(instance=StandardPage(title=i))
```

Go to: http://127.0.0.1:8000/admin/pages/3/edit/ - and use the multi chooser. It should fix the following issues:

- When searching, the chooser on the left hand side disappears
- When searching, a "Parent" column appears, when you click this it breaks the multi chooser experience a bit

Screenshot 2024-06-12 at 16 53 23
Screenshot 2024-06-12 at 16 53 31

It should now look like:

Screenshot 2024-06-12 at 16 55 43

### CODE REVIEW

File: wagtail/admin/templates/wagtailadmin/chooser/browse.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
     {% trans "Choose a page" as choose_str %}
 {% endif %}
 
-{% include "wagtailadmin/shared/header.html" with title=choose_str subtitle=page_type_names|join:", " search_url="wagtailadmin_choose_page_search" query_parameters="page_type="|add:page_type_string icon="doc-empty-inverse" search_disable_async=True %}
+{% include "wagtailadmin/shared/header.html" with title=choose_str subtitle=page_type_names|join:", " search_url="wagtailadmin_choose_page_search" query_parameters="page_type="|add:page_type_string|add:"&multiple="|add:is_multiple_choice icon="doc-empty-inverse" search_disable_async=True %}
```
Comment (tomkins) [inline wagtail/admin/templates/wagtailadmin/chooser/browse.html:8]: Note - part of me feels a little bit awkward at just continuing with this logic in the template, happy to refactor this into Python if needed.

**Comment by gasman** [comment]
Thanks @tomkins! Some initial thoughts:

* #11955 appears to be the issue being fixed here. I think #10754 is something different - that's about the state of the checkbox not being remembered while navigating around the chooser, as a general user-experience thing applicable to all chooser types. The description in #10451 is a bit too vague for me to tell which of #11955 or #10754 it's referring to ("make the checkbox select dissapear" could mean the checkbox losing its selected state or disappearing entirely), but it's a duplicate either way :-)
* The underlying problem with the "Parent" column is that it's not preserving any of the URL parameters. This is most obviously seen here with the `multiple` parameter, but it also affects `page_type` - for example, if after selecting some related pages you select "Choose another page" to open the single-select interface, then run a search and click on a "Home" parent link, the subsequent page will allow you to select "Home" when it shouldn't. I'd rather that we fixed that underlying problem rather than just removing the parent column in multiple-select mode, since at the moment this is introducing a difference between the two modes for no real reason.
* I agree it would be better to do the URL construction on the Python side rather than making the template code more complex. If I remember correctly, there's already some fairly extensive code in ChooserViewSet for determining which URL variables to pass on in links, which is used for the pagination (and evidently the search bar in the snippet chooser, because that doesn't suffer from this issue...) - I think that could probably be re-used here for both the search bar and the 'parent' link. Happy to take a look at this later (if you don't get to it first)...

**Comment by tomkins** [comment]
Thanks for the initial feedback - will get onto this in a couple of days

**Comment by dopry** [comment]
@tomkins any chance you have time to keep this moving. I know my team would be very happy to see a MultipleChooserPanel that worked. It's been unusable since it was shipped in May 2023.

**Comment by tomkins** [comment]
Will get back on this one shortly (apologies) - as yes, it's definitely an issue I'd like to see squashed!

**Comment by gasman** [comment]
Superseded by #12643

**Pull request closed without merging**